### PR TITLE
TODO squashing pt 1

### PR DESCRIPTION
**Pull request by Mazey**
## Status

(pick one)

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

- TODO 1: TTH warmup now uses updated (CTF) round start sound
- TODO 2: Rather than nothing, show skull when dead. Additionally to this TODO, updated class icons to better ones and show different (inactive-ish) icons when dead
- TODO 3: Bomb arrows now flash red when their velocity starts increasing.

Screenshot for the scoreboard change:
image

\#TodoADayChallenge

### CODE REVIEW

**Comment by Mazey** [comment]
Skull scale now matches class icons.

File: Entities/Items/Projectiles/Arrow.as [hunk 1]
```diff
@@ -155,6 +155,24 @@ void onTick(CBlob@ this)
 
 			processSticking = false;
 		}
+		
+		if (arrowType == ArrowType::bomb)
+		{
+			if (!this.exists("old velocity") || this.get_f32("old velocity") > shape.vellen)
```
Comment (Vam-Jam) [inline Entities/Items/Projectiles/Arrow.as:161]: would it not be easier to compare this.getVelocity and this.getOldVelocity

Comment (Vam-Jam) [reply Entities/Items/Projectiles/Arrow.as:161]: (also of note, there's also oldvellen var, but its not been exposed to angelscript, could be worth doing that at some point)

File: Entities/Items/Projectiles/Arrow.as [hunk 2]
```diff
@@ -155,6 +155,24 @@ void onTick(CBlob@ this)
...
+				this.set_f32("old velocity", shape.vellen);
+				return;
+			}
+
+			CSprite@ sprite = this.getSprite();
+
+			if (sprite !is null)
+			{
+				sprite.animation.time = 2;
+				sprite.animation.loop = true;
```
Comment (Vam-Jam) [inline Entities/Items/Projectiles/Arrow.as:172]: wont you need to turn this off if speed decreases

File: Rules/CommonScripts/ScoreboardRender.as [hunk 3]
```diff
@@ -174,24 +175,45 @@ float drawScoreboard(CPlayer@ localplayer, CPlayer@[] players, Vec2f topleft, CT
...
+			classIndex = 0;
+
+			// why are player-scoreboard functions hardcoded
+			// after looking into it let's not bother moving it to scripts for now
+			classIndex = p.getScoreboardFrame();
+
+			// knight is 3 but should be 0 for this texture
+			// fyi it's pure coincidence builder and archer are already a match
+			if (classIndex == 3) 
+			{
```
Comment (Vam-Jam) [inline Rules/CommonScripts/ScoreboardRender.as:202]: could remove brackets

File: Rules/CommonScripts/ScoreboardRender.as [hunk 4]
```diff
@@ -174,24 +175,45 @@ float drawScoreboard(CPlayer@ localplayer, CPlayer@[] players, Vec2f topleft, CT
...
+			// knight is 3 but should be 0 for this texture
+			// fyi it's pure coincidence builder and archer are already a match
+			if (classIndex == 3) 
+			{ 
+				classIndex = 0;
+			}
+			
+			classIconSize = Vec2f(16, 16);
+
+			if (b is null) { // player dead
```
Comment (Vam-Jam) [inline Rules/CommonScripts/ScoreboardRender.as:208]: bracket on a new line >:(

File: Rules/CommonScripts/ScoreboardRender.as [hunk 5]
```diff
@@ -215,25 +237,25 @@ float drawScoreboard(CPlayer@ localplayer, CPlayer@[] players, Vec2f topleft, CT
 
 		}
 
-		//head icon
+		// head icon
 
-		//TODO: consider maybe the skull emoji for dead players?
-		int headIndex = 0;
-		string headTexture = "";
+		string headTexture = "playercardicons.png";
```
Comment (Vam-Jam) [inline Rules/CommonScripts/ScoreboardRender.as:242]: make this a const string at the top of the file

**Comment by Mazey** [comment]
- TODO 4: Add leave particles when trees die

\#TodoADayChallenge

**Comment by eps0003** [comment]
Each of these should be their own PR instead of clumping them into one.

File: Entities/Natural/Trees/LogsOnDeath.as [hunk 6]
```diff
@@ -20,24 +20,27 @@ void onDie(CBlob@ this)
...
-			pos = this.getPosition() + (segment.start_pos + segment.end_pos) / 2.0f;
-			pos.y -= 4.0f; // TODO: fix logs spawning in ground
 			CBlob@ log = server_CreateBlob("log", this.getTeamNum(), pos);
 			if (log !is null)
-			{
 				log.setAngleDegrees(fall_angle);
-			}
 		}
+		
+		if (XORRandom(100) < 20)
```
Comment (eps0003) [inline Entities/Natural/Trees/LogsOnDeath.as:34]: `XORRandom(5) == 0` makes it clearer its a 1 in 5 chance.

File: Entities/Natural/Trees/LogsOnDeath.as [hunk 7]
```diff
@@ -20,24 +20,27 @@ void onDie(CBlob@ this)
...
+
+		ParticleAnimated("Entities/Effects/Sprites/Leaves", pos + Vec2f(6-XORRandom(12), 0), Vec2f(0,-0.75f), 0.0f, 0.5f, 4, 0.1f, false);
 	}
 
-	//TODO LEAVES PARTICLES
-	//ParticleAnimated( "Entities/Effects/leaves", pos, Vec2f(0,-0.5f), 0.0f, 1.0f, 2+XORRandom(4), 0.2f, false );
-	//for (int i = 0; i < this.getSprite().getSpriteLayerCount(); i++) { // crashes
-	//    ParticlesFromSprite( this.getSprite().getSpriteLayer(i) );
-	//}
+	for (int i = 0; i < this.getSprite().getSpriteLayerCount(); i++) {
```
Comment (eps0003) [inline Entities/Natural/Trees/LogsOnDeath.as:40]: Opening curly bracket should go on a new line to match code style. Also, you should move `this.getSprite().getSpriteLayerCount()` so it's not getting the sprite and the layer count every iteration of the loop.

**Comment by AsuMagic** [comment]
> TODO 1: TTH warmup now uses updated (CTF) round start sound
>
> TODO 2: Rather than nothing, show skull when dead. Additionally to this TODO, updated class icons to better ones and show different (inactive-ish) icons when dead

LGTM

> TODO 3: Bomb arrows now flash red when their velocity starts increasing.

It looks good visually, but I don't know if it makes much sense? It might mislead new players into thinking that bombarrows have some kind of fuse when shot long distance (as if they were bombs), when they don't.

> TODO 4: Add leave particles when trees die

Would need to look more refined IMO, the leaves particles turn into pixel particles with high velocity for some reason (relevant to #1624. Also pine trees emit a ton of odd looking pixel particles on death here but bush trees don't.

**Comment by AsuMagic** [comment]
Maybe the bombarrows should always just flash in that case?

**Comment by Mazey** [comment]
> Maybe the bombarrows should always just flash in that case?

I made it this way to try to "fulfill" the TODO as imposed by Geti or MM, but I see your point. I've changed it to always flash.

You can merge without the particle stuff, I don't think it'll be feasible to rewrite it considering the issues with particle in general as you mention.

**Review by AsuMagic — APPROVED** [review]

**Pull request merged**